src/Propulsion.SqlStreamStore/SqlStreamStoreSink.fs

```diff
@@ -0,0 +1,142 @@
+﻿namespace Propulsion.SqlStreamStore
+
+open Equinox.SqlStreamStore
+open Propulsion
+open Propulsion.Streams
+open Propulsion.Streams.Internal // Helpers
+open Serilog
+open System.Collections.Generic
+open System
+open System.Threading
+
+[<AutoOpen>]
+module private Impl =
+    let inline mb x = float x / 1024. / 1024.
+
+module Internal =
+
+    [<AutoOpen>]
+    module Writer =
+        type EventStoreContext() = class end
+
+        type [<NoComparison; NoEquality>] Result =
+            | Ok of updatedPos : int64
+            | ConflictUnknown
+
+        let logTo (log : ILogger) (res : FsCodec.StreamName * Choice<(int * int) * Result, (int * int) * exn>) =
+            match res with
+            | stream, (Choice1Of2 (_, Ok pos)) ->
+                log.Information("Wrote     {stream} up to {pos}", stream, pos)
+            | stream, (Choice1Of2 (_, ConflictUnknown)) ->
+                log.Information("ConflictUnknown   {stream}", stream)
+            | stream, (Choice2Of2 (_, exn)) ->
+                log.Warning(exn,"Writing   {stream} failed, retrying", stream)
+            
+        let write (log : ILogger) (context : Context) stream span = async {
+            log.Debug("Writing {s}@{i}x{n}", stream, span.index, span.events.Length)
+            let! res = context.Sync(log, stream, span.index - 1L, span.events |> Array.map (fun x -> x :> _))
+            let ress =
+                match res with
+                | GatewaySyncResult.Written (Token.Unpack pos') ->
+                    Ok (pos'.pos.streamVersion + 1L)
+                | GatewaySyncResult.ConflictUnknown ->
```

Comment:
need to build logic to ascertain the write position - that's also be useful in Equinox.SqlStreamStore, so my thinking would be to do it there, and let this simply use the position (i.e. make ConflictUnknown bear the position too)

an alternate resolution of this is to remove this Sink entirerely for now - there's already a lot of value from having the reader working; I'd argue that's even higher value than being able to use SSS as a sink